src/Period.php

```diff
@@ -436,6 +438,15 @@ public function getPrecisionMask(): int
         return $this->precisionMask;
     }
 
+    public function getIterator()
+    {
+        return new DatePeriod(
+            $this->getIncludedStart(),
+            $this->interval,
+            $this->getIncludedEnd()->add($this->interval)
```

Comment:
Am I correct to assume that `\DatePeriod` always uses an inclusive-exclusive range when iterating? so `[start, end)` ? I think we could use `$this->end`, instead of adding a unit to the included and.

Reply:
I tried changing it to `$this->end` and the tests broke (the iterator had always one item less than expected). I must admit that I haven't completely grasped the difference between `getIncluded(Start|End)` and `get(Start|End)` yet, so I looked at the other methods and they mostly used the `getIncludedEnd()` 😅

Reply:
Thinking about it, there is a difference indeed. `getIncludedEnd` + 1 is the right one to use! `getEnd` will return the end date the user specified when creating the period, but that can either be included or excluded, based on how the period was constructed. So scratch my comment, you were right!